### PR TITLE
Remakefile

### PR DESCRIPTION
**Pull request by jsadler2**
remake workflow

### CODE REVIEW

**Comment by jsadler2** [comment]
image

**Comment by jsadler2** [comment]
@aappling-usgs or @jread-usgs  - I think I've done this all correctly. But maybe I jumped to a "proper" yml before the bot could give me feedback. Do you guys have any feedback on my code?

File: 3_plot/src/plot.R [hunk 1]
```diff
@@ -0,0 +1,46 @@
+library(whisker)
+
+plot_figure <- function(summary_file, outfile, diagnostic_file){
+png(file = file.path(outfile), width = 8, height = 10, res = 200, units = 'in')
```
Comment (aappling-usgs) [inline 3_plot/src/plot.R]: could you do a cmd-a, cmd-i on this file (and any others that need it) to use RStudio's default indentation? would make it easier to read.

File: remake.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - 2_process/src/process.R
+  # trying to trick the bot since I've already done it right
+  - 3_plot/src/plot.R
+  - 4_diagnostics/src/diagnostics.R
+
+targets:
+  all:
+    depends:
+      - 3_plot/out/figure_1.png
+      - 4_diagnostics/out/model_diagnostic_text.txt
```
Comment (aappling-usgs) [inline remake.yml:14]: There are a few file targets that won't currently be built using the default target (i.e., if you just call `scmake()` with no arguments) - could you add those here?

File: remake.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+sources:
+  - 1_fetch/src/fetch.R
+  - 2_process/src/process.R
+  - 3_plot/src/plot.R
+  - 4_diagnostics/src/diagnostics.R
+
```
[the remark this answers is not in the document]
Comment (aappling-usgs) [reply remake.yml:7]: you've done this

File: remake.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+targets:
+  all:
+    depends:
```
[the remark this answers is not in the document]
Comment (aappling-usgs) [reply remake.yml:10]: done

File: remake.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      - 3_plot/out/figure_1.png
+      - 4_diagnostics/out/model_diagnostic_text.txt
+
```
[the remark this answers is not in the document]
Comment (aappling-usgs) [reply remake.yml:15]: done

File: remake.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  1_fetch/out/model_RMSEs.csv:
+    command: pull_data(outfile = target_name)
```
[the remark this answers is not in the document]
Comment (aappling-usgs) [reply remake.yml:17]: done

**Review by aappling-usgs — CHANGES_REQUESTED** [review]
Yeah, I think you're interpreting the bot's comments correctly - they're in the wrong places because you'd already edited the remake.yml. But I think you got those comments resolved anyway.

I've added a couple of comments about completing the default target and using RStudio to get consistent indentation throughout.

File: 1_fetch/src/fetch.R [hunk 7]
```diff
@@ -0,0 +1,16 @@
+library(dplyr)
+library(readr)
+library(stringr)
+library(sbtools)
+
+pull_data <- function(outfile){
+  mendota_file <- file.path(outfile)
```
Comment (aappling-usgs) [inline 1_fetch/src/fetch.R]: file.path is just a string operation (doesn't change outfile if that's the only argument), so it's unnecessary now

File: 2_process/src/process.R [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+          model_type == 'pb' ~ '#1b9e77',
+          model_type == 'dl' ~'#d95f02',
+          model_type == 'pgdl' ~ '#7570b3'
+        ), pch = case_when(
+          model_type == 'pb' ~ 21,
+          model_type == 'dl' ~ 22,
+          model_type == 'pgdl' ~ 23
+        ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+
+    readr::write_csv(eval_data, path = file.path(outfile))
```
Comment (aappling-usgs) [inline 2_process/src/process.R]: again here file.path isn't needed

**Review by aappling-usgs — COMMENTED** [review]
one other teeny comment (twice) about R's `file.path()` function

File: remake.yml (hunk 2 above)
Comment (jsadler2) [reply remake.yml:14]: I think I've resolved this now. I think I just did these b/c I knew the other products depended on these.

File: 3_plot/src/plot.R (hunk 1 above)
Comment (jsadler2) [reply 3_plot/src/plot.R]: done

File: 1_fetch/src/fetch.R (hunk 7 above)
Comment (jsadler2) [reply 1_fetch/src/fetch.R]: ohh. it's like `os.path.join` in python. Thanks.

Comment (jsadler2) [reply 1_fetch/src/fetch.R]: done

File: 2_process/src/process.R (hunk 8 above)
Comment (jsadler2) [reply 2_process/src/process.R]: done

**Pull request merged**